--- a/schedule.c
+++ b/schedule.c
@@ -6,7 +6,12 @@ static void recv_ifevent_netlink_sk(void)
 	char buf[4096]; //test this with a very small value !!
 	struct iovec iov = {buf, sizeof(buf)};
 	struct sockaddr_nl sa;
-	struct msghdr msg = {(void *)&sa, sizeof(sa), &iov, 1, NULL, 0, 0};
+	struct msghdr msg = {
+		.msg_name = (void *)&sa,
+		.msg_namelen = sizeof(sa),
+		.msg_iov = &iov,
+		.msg_iovlen = 1,
+	};
 
 	while (recvmsg(ifevent_sk, &msg, 0) > 0)
 		;
